Signal/src/ViewControllers/PhotoLibrary/ImagePickerController.swift

```diff
@@ -12,21 +12,25 @@ protocol ImagePickerControllerDelegate {
 }
 
 @objc(OWSImagePickerGridController)
-class ImagePickerGridController: UICollectionViewController, PhotoLibraryDelegate {
+class ImagePickerGridController: UICollectionViewController, PhotoLibraryDelegate, PhotoCollectionPickerDelegate {
 
     @objc
     weak var delegate: ImagePickerControllerDelegate?
 
     private let library: PhotoLibrary = PhotoLibrary()
-    private let libraryAlbum: PhotoLibraryAlbum
-
-    var availableWidth: CGFloat = 0
+    private var photoCollection: PhotoCollection
+    private var photoCollectionContents: PhotoCollectionContents
+    private let photoMediaSize = PhotoMediaSize()
```

Comment:
I tried to reuse as much of the image picker as I could, e.g. the `PhotoLibrary` class, etc.  This involved a few changes to generalize this logic.  

In this case, we were previously hanging the desired thumbnail size on the album/collection contents. 
 I've created a new `PhotoMediaSize` struct for hanging the desired thumbnail size.  The image picker can dynamically update its behavior and the album picker can use a fixed value.